src/test/java/dev/streamx/cli/publish/PublishCommandTest.java

```diff
@@ -0,0 +1,172 @@
+import static org.apache.hc.core5.http.HttpStatus.SC_BAD_REQUEST;
+import static org.apache.hc.core5.http.HttpStatus.SC_OK;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.github.tomakehurst.wiremock.client.WireMock;
+import com.github.tomakehurst.wiremock.common.Json;
+import com.github.tomakehurst.wiremock.junit5.WireMockExtension;
+import dev.streamx.clients.ingestion.impl.FailureResponse;
+import dev.streamx.clients.ingestion.publisher.PublisherSuccessResult;
+import io.quarkus.test.junit.main.LaunchResult;
+import io.quarkus.test.junit.main.QuarkusMainLauncher;
+import io.quarkus.test.junit.main.QuarkusMainTest;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+@QuarkusMainTest
+public class PublishCommandTest {
+  private static final String CHANNEL = "pages";
+  private static final String BAD_REQUEST_CHANNEL = "bad-request-channel";
+  private static final String KEY = "index.html";
+  private static final String DATA = """
+        {"content": {"bytes": "<h1>Hello World!</h1>"}}""";
+
+  @RegisterExtension
+  static WireMockExtension wm = WireMockExtension.newInstance()
+      .options(wireMockConfig().dynamicPort()).configureStaticDsl(true).build();
+
+  @BeforeEach
+  void setup() {
+    initializeWiremock();
+  }
+
+  @Test
+  public void shouldRejectIllegalIngestionUrl(QuarkusMainLauncher launcher) {
+    // given
+    String invalidIngestionUrl = "hattetepe:///in valid";
+
+    // when
+    LaunchResult result = launcher.launch("publish",
+        "--ingestion-url=" + invalidIngestionUrl,
+        "--data=" + DATA,
+        CHANNEL, KEY);
+
+    // then
+    assertThat(result.exitCode()).isNotZero();
+    assertThat(result.getErrorOutput()).contains("Illegal character in path at index 15");
+  }
+
+  @Test
+  public void shouldRejectInvalidHostInIngestionUrl(QuarkusMainLauncher launcher) {
+    // given
+    String invalidIngestionUrl = "hattetepe:///invalid";
+
+    // when
+    LaunchResult result = launcher.launch("publish",
+        "--ingestion-url=" + invalidIngestionUrl,
+        "--data=" + DATA,
+        CHANNEL, KEY);
+
+    // then
+    assertThat(result.exitCode()).isNotZero();
+    assertThat(result.getErrorOutput()).contains("URI does not specify a valid host name");
+  }
+
+  @Test
+  public void shouldHandleBadRequestFromRestIngestionApi(QuarkusMainLauncher launcher) {
+    // when
+    LaunchResult result = launcher.launch("publish",
+        "--ingestion-url=" + getIngestionUrl(),
+        "--data=" + DATA,
+        BAD_REQUEST_CHANNEL, KEY);
+
+    // then
+    assertThat(result.exitCode()).isNotZero();
+    assertThat(result.getErrorOutput()).contains("Ingestion client exception. Message: Publication Ingestion REST endpoint known error. Code: INVALID_PUBLICATION_PAYLOAD. Message: Error message");
```

Comment:
Why do we have "Message:" twice in this output?

Reply:
Yes, double "Message" will be fixed.

However Messages from StreamxClientExceptions requires more work that will be done within [DXP-707](https://teamds.atlassian.net/browse/DXP-707)